curvefs/src/client/rpcclient/task_excutor.cpp

```diff
@@ -0,0 +1,220 @@
+        uint64_t nextTimeout = 0;
+        uint64_t retriedTimes = task_->retryTimes;
+        bool leaderMayChange =
+            metaCache_->IsLeaderMayChange(task_->poolID, task_->copysetID);
+
+        if (retriedTimes < opt_.minRetryTimesForceTimeoutBackoff &&
+            leaderMayChange) {
+            nextTimeout = opt_.rpcTimeoutMS;
+        } else {
+            nextTimeout = TimeoutBackOff();
+        }
+
+        task_->rpcTimeoutMs = nextTimeout;
+        LOG(WARNING) << "rpc timeout, next timeout = " << nextTimeout
+                     << task_->TaskContextStr();
+        return true;
+    }
+
+    // over loead
+    if (retCode == MetaStatusCode::OVERLOAD) {
+        uint64_t nextsleeptime = OverLoadBackOff();
+        LOG(WARNING) << "metaserver overload, sleep(us) = " << nextsleeptime
+                     << ", " << task_->TaskContextStr();
+        bthread_usleep(nextsleeptime);
+        return true;
+    }
+
+    if (!task_->retryDirectly) {
+        bthread_usleep(opt_.retryIntervalUS);
+    }
+    return true;
+}
+
+bool TaskExcutor::GetTarget() {
+    if (!metaCache_->GetTarget(task_->fsID, task_->inodeID, &task_->targetID,
+                               &task_->targetAddr, &task_->poolID,
+                               &task_->copysetID, &task_->partitionID,
+                               &task_->applyIndex)) {
+        LOG(ERROR) << "fetch target for task fail, " << task_->TaskContextStr();
+        return false;
+    }
+    return true;
+}
+
+int TaskExcutor::ExcuteTask() {
+    // fetch target
+    if (!GetTarget()) {
+        return -1;
+    }
+
+    // get channel and send rpc
+    auto channel =
+        channelManager_->GetOrCreateChannel(task_->targetID, task_->targetAddr);
+    if (channel == nullptr) {
+        return -EHOSTDOWN;
+    }
+
+    brpc::Controller cntl;
+    cntl.set_timeout_ms(task_->rpcTimeoutMs);
+    return task_->rpctask(task_->poolID, task_->copysetID, task_->partitionID,
+                          task_->applyIndex, channel.get(), &cntl);
+}
+void TaskExcutor::OnSuccess() {
+    metaCache_->UpdateApplyIndex(task_->poolID, task_->copysetID,
+                                 task_->applyIndex);
+}
+void TaskExcutor::OnCopysetNotExist() { RefreshLeader(); }
+
+void TaskExcutor::OnOverLoad() {}
+void TaskExcutor::OnReDirected() { RefreshLeader(); }
+
+void TaskExcutor::RefreshLeader() {
+    MetaserverID oldTarget = task_->targetID;
+    bool ok = metaCache_->GetTarget(
+        task_->fsID, task_->inodeID, &task_->targetID, &task_->targetAddr,
+        &task_->poolID, &task_->copysetID, &task_->partitionID,
+        &task_->applyIndex, true);
+    LOG_IF(WARNING, !ok) << "Refresh leader failed, task:"
+                         << task_->TaskContextStr();
+    task_->retryDirectly = (oldTarget != task_->targetID);
+}
+
+
+uint64_t TaskExcutor::OverLoadBackOff() {
+    uint64_t curpowTime = std::min(task_->retryTimes, uint32_t(1));
```

Comment:
std::min ?   curpowTime only can be 0 or 1 ?

Reply:
fix